src/App.tsx

```diff
@@ -1,14 +1,52 @@
 /* eslint-disable max-len */
 import React from 'react';
+import { useState, useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [selectTodo, setSelectTodo] = useState<Todo | null>(null);
+  const [selectFilter, setSelectFilter] = useState('all');
+  const [query, setQuery] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    getTodos()
+      .then(data => {
+        setTodos(data);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
```

Comment:
Consider adding error handling for the `getTodos` function. Currently, if the promise is rejected, there is no way to handle the error, which could lead to unexpected behavior. You might want to add a `.catch()` block to handle any potential errors.